### PR TITLE
Bug axon zerobytes

### PR DESCRIPTION
**Pull request by vEpiphyte**
Add tests for eatfd / eatbytes which have zero bytes objects
Fix eatfd by ensuring that we chunk the first blob of bytes we consume and then pull in a second read prior to entering the while loop.
Fix eatbytes by changing s_common.chunks() to handle empty sequences.
Add a cortex test to ensure that getRowsByIdens() returns an empty list when presented with a empty iterable of idens.

### CODE REVIEW

File: synapse/tests/test_common.py [hunk 1]
```diff
@@ -57,3 +57,47 @@ def test_common_listdir(self):
...
+        self.eq(parts, [[]])
+
+        parts = [chunk for chunk in chunks('', 10000)]
+        self.eq(parts, [''])
+
+        parts = [chunk for chunk in chunks([1, 2, 3, 4, 5], 2)]
+        self.eq(parts, [[1, 2], [3, 4], [5]])
+
+        # set is unslicable
+        tpass = False
```
Comment (vertexmc) [inline synapse/tests/test_common.py]: Can you just use `self.raises(...` instead of these `tpass`es?

Comment (vEpiphyte) [reply synapse/tests/test_common.py]: Have to use assertRaises() directly as a context manager, since we can't call chunks() directly since it is a generator that needs to be consumed in order to have the exception thrown.  I replaced this with assertRaises.
